SetGame/SetGame/SetGameModel.swift

```diff
@@ -0,0 +1,274 @@
+        return comparison(self.selectedCards[0].color, self.selectedCards[1].color, self.selectedCards[2].color) &&
+            comparison(self.selectedCards[0].filling, self.selectedCards[1].filling, self.selectedCards[2].filling) &&
+            comparison(self.selectedCards[0].shape, self.selectedCards[1].shape, self.selectedCards[2].shape) &&
+            comparison(self.selectedCards[0].shapeCount, self.selectedCards[1].shapeCount, self.selectedCards[2].shapeCount)
+    }
+    
+    //return 3 indexes of existing set, or nil if not exist
+    mutating func getHint() -> (Int, Int, Int)? {
+        //if selected card already a match - first replace selected cards
+        if self.choosenCardsState == .match {
+            let indexes = self.clearSelectedSet()
+            self.addCardsToBoard(at: indexes)
+        }
+        //reduce score by one
+        self.score -= 1
+        return self.findSet()
+    }
+    
+    //return 3 indexes of existing set, or nil if not exist
+    func findSet() -> (Int, Int, Int)? {
+        for var i in 0..<cardOnBoard.count {
+            for var j in i+1..<cardOnBoard.count {
+                if let k = thirdCardForSetExist(index1: i, index2: j) {
+                    return (i, j, k)
+                }
+            }
+        }
+        return nil
+    }
+    
+    //check if board have third card to complete the given 2 cards to a set, and return the index
+    func thirdCardForSetExist(index1: Int, index2: Int) -> Int? {
+        let card1 = self.cardOnBoard[index1]
+        let card2 = self.cardOnBoard[index2]
+        //if cards are nill print error
+        guard card1 != nil && card2 != nil else {
+            return nil
+        }
+        //create complete card for set
+        let shape = thirdPropertyForSet(prop1: card1.shape, prop2: card2.shape)
+        let color = thirdPropertyForSet(prop1: card1.color, prop2: card2.color)
+        let shapeCount = thirdPropertyForSet(prop1: card1.shapeCount, prop2: card2.shapeCount)
+        let filling = thirdPropertyForSet(prop1: card1.filling, prop2: card2.filling)
+        let card3 = Card(shape: shape, color: color, shapeCount: shapeCount, filling: filling)
+        
+        return self.cardOnBoard.contains(card3) ? self.cardOnBoard.firstIndex(of: card3) : nil
+    }
+    
+    //return third property for given 2 to complete a set
+    private func thirdPropertyForSet(prop1: CardProperty, prop2: CardProperty) -> CardProperty
+    {
+        //return the same property for "same" set
+        if prop1 == prop2 {
+            return prop1
+        }
+        //find the third property for "different" set
+        else {
+            let propertyValues = CardProperty.allCases
+            for p in propertyValues {
+                if p != prop1 && p != prop2 {
+                    return p
+                }
+            }
+        }
+        return prop1
```

Comment:
This line should never be reached.. please note it's possible to implement the same logic with 2 `return` statements instead of 3